From a09f61d766c9bef07350ca68730ef0f2ebf102bb Mon Sep 17 00:00:00 2001
From: Dummy <dummy@gmail.com>
Date: Sat, 29 Aug 2026 22:44:18 +0000
Subject: [PATCH] arm64: dts: rk3399-pinephone-pro: Keep modem regulators
 always-on

The vcc_4g and vcc_4g_5v regulators power the EG25 modem. Mark them
always-on so the modem is powered at boot when using eg25-manager.
---
 arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index bb72e41366dc..14b1e4b405a0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -291,6 +291,7 @@ vcc_4g_5v: regulator-vcc-4g-5v {
 		regulator-min-microvolt = <5000000>;
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&vcc5v0_sys>;
+       regulator-always-on;
 	};
 
 	vcc_4g: regulator-vcc-4g {
@@ -303,6 +304,7 @@ vcc_4g: regulator-vcc-4g {
 		regulator-min-microvolt = <3800000>;
 		regulator-max-microvolt = <3800000>;
 		vin-supply = <&vcc_sys>;
+       regulator-always-on;
 	};
 
 	vcc1v8_codec: regulator-vcc1v8-codec {
-- 
2.55.0

